CodingGuidelines.md

```diff
@@ -0,0 +1,168 @@
+# Coding Guidelines
+
+This document outlines the recommended coding guidelines for the Mixed Reality Toolkit.  The majority of these suggestions follow the [recommended standards from MSDN](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions).
+
+## Spaces vs Tabs
+Please be sure to use 4 spaces when contributing to this project.
+
+## Naming Conventions
+
+Always use `PascalCase` for public properties, and `camelCase` for private properties and fields.
+> The only exception to this is for data structures that require the fields to be serialized by the `JsonUtility`.
```

Comment:
an example of the exception would be useful

Reply:
Agreed, I'm going to do a PR on @StephenHodgson branch to add my thoughts (esp as the next section actually breaks this principle :P)